### PR TITLE
Capture traceability on failure and fix traceability call issue

### PR DESCRIPTION
**Pull request by ajinkya599**

### CODE REVIEW

File: src/run.ts [hunk 1]
```diff
@@ -4,12 +4,13 @@ import * as path from 'path';
...
 
 import { downloadKubectl, getStableKubectlVersion } from "./utilities/kubectl-util";
-import { getExecutableExtension, isEqual } from "./utilities/utility";
+import { getDeploymentConfig, getExecutableExtension, isEqual, isValidAction } from "./utilities/utility";
 
 import { Kubectl } from './kubectl-object-model';
 import { deploy } from './utilities/strategy-helpers/deployment-helper';
 import { promote } from './actions/promote';
 import { reject } from './actions/reject';
+import * as AzureTraceabilityHelper from "./utilities/../traceability/azure-traceability-helper";
```
Comment (Ganeshrockz) [inline src/run.ts:13]: Looks like the path can directly begin with `./traceability`

File: src/traceability/azure-traceability-helper.ts [hunk 2]
```diff
@@ -97,7 +98,7 @@ function getDeploymentResourceUri(aksResourceContext: AksResourceContext): strin
   return `${aksResourceContext.managementUrl}subscriptions/${aksResourceContext.subscriptionId}/resourceGroups/${aksResourceContext.resourceGroup}/providers/Microsoft.Devops/deploymentdetails/${deploymentName}?api-version=2020-12-01-preview`;
 }
 
-async function createDeploymentReport(kubectl: Kubectl, deploymentConfig: DeploymentConfig): Promise<DeploymentReport> {
+async function createDeploymentReport(kubectl: Kubectl, deploymentConfig: DeploymentConfig, status: string): Promise<DeploymentReport> {
```
Comment (Josh-01) [inline src/traceability/azure-traceability-helper.ts:101]: Should we make 'status' - > 'runStatus' or 'deployStatus' ?

File: src/traceability/azure-traceability-helper.ts [hunk 3]
```diff
@@ -120,7 +121,7 @@ async function createDeploymentReport(kubectl: Kubectl, deploymentConfig: Deploy
   });
 
   const deploymentReport: DeploymentReport = new DeploymentReport();
-  await deploymentReport.initialize(InputParameters.githubToken, artifacts, 'succeeded', resource);
+  await deploymentReport.initialize(InputParameters.githubToken, artifacts, status, resource);
```
Comment (Josh-01) [inline src/traceability/azure-traceability-helper.ts:124]: Is the expected string literal 'succeeded' or 'success' ? (It is set as success in run())

**Review by Ganeshrockz — APPROVED** [review]

File: src/traceability/azure-traceability-helper.ts (hunk 2 above)
Comment (Josh-01) [inline src/traceability/azure-traceability-helper.ts:101]: Should we make 'status' - > 'runStatus' or 'deployStatus' ?

File: src/traceability/azure-traceability-helper.ts (hunk 3 above)
Comment (Josh-01) [inline src/traceability/azure-traceability-helper.ts:124]: Is the expected string literal 'succeeded' or 'success' ? (It is set as 'success' in run())

File: src/run.ts (hunk 1 above)
Comment (ajinkya599) [reply src/run.ts]: Changed.

File: src/traceability/azure-traceability-helper.ts (hunk 3 above)
Comment (ajinkya599) [reply src/traceability/azure-traceability-helper.ts]: It seems `success` as per the example. Asked on the channel.

File: src/traceability/azure-traceability-helper.ts (hunk 2 above)
Comment (ajinkya599) [reply src/traceability/azure-traceability-helper.ts]: Changed to `runStatus`.

**Review by Josh-01 — APPROVED** [review]

File: src/traceability/azure-traceability-helper.ts (hunk 3 above)
Comment (ajinkya599) [reply src/traceability/azure-traceability-helper.ts]: Going ahead with `success/failure`. Can take any change incrementally.

**Pull request merged**